rsa.d

```diff
@@ -0,0 +1,491 @@
+// Author: Adam Williams. 2016-2017.
+
+// Implements evp encrypting and decrypting functionality using only RSA.
+// Implements sealing and opening envelope functionality using a combination of RSA and AES.
+// This file depends on the openssl library. Please note: Currently no support for the Botan library.
+
+module secured.rsa;  
+
+import core.memory;
+
+version(OpenSSL)
+{
+import deimos.openssl.evp;
+import deimos.openssl.rand;
+import deimos.openssl.pem;
+import deimos.openssl.bio;
+import deimos.openssl.rsa;
+import deimos.openssl.engine;
+}
+version(Botan)
+{
+	static assert(0, "Botan is not support in module secured.rsa.");
+}
+
+import secured.random;
+import secured.util;
+
+// CONSTANTS
+const ulong RSA_KEYLEN = 2048;
+
+// ----------------------------------------------------------
+
+public class RSA
```

Comment:
Please add Sign and Verify, these are critical operations for secure long-term data storage. Implementation would be roughly the same as in EllipticCurve.